api/enki.py

```diff
@@ -221,8 +221,8 @@ def reaper_request(self, identifier):
             data = json.loads(response.content)
             self.log.debug ("Keeping existing book: " + str(identifier))
         except:
-            # Get the license pool for the ID, but make sure it's the one belonging to Enki
-            pool = [x for x in identifier.licensed_through if x.data_source.name == DataSource.ENKI][0]
+            # Get this collection's license pool for this identifier.
+            pool = identifier.licensed_through_collection(self.collection)
```

Comment:
If two libraries have two Enki collections it's possible for the old code to return more than one LicensePool, and it might not be the one we're supposed to reap.